### PR TITLE
Add documentation - Advanced - API Keys

### PR DESCRIPTION
**Pull request by Aschen**
https://deploy-preview-1878--doc-kuzzle.netlify.app/guides/advanced/2-api-keys

### CODE REVIEW

File: doc/2/guides/advanced/2-api-keys/index.md [hunk 1]
```diff
@@ -2,9 +2,121 @@
...
-description: tbd
+description: Manage API keys 
 order: 200
 ---
 
 # API Keys
 
+Kuzzle allows to create API keys to **authenticate users without using an authentication strategy** and the [auth:login](/core/2/references/controllers/auth/login) action.
+
+An API key is associated with a standard [authentication token](/core/2/guides/main-concepts/5-authentication#authentication-token) that can then be used to [authenticate users] to the Kuzzle API.
```
Comment (jenow) [inline doc/2/guides/advanced/2-api-keys/index.md:13]: There is a link missing at `[authenticate users]`

**Review by jenow — CHANGES_REQUESTED** [review]

File: doc/2/guides/advanced/1-secrets-vault/index.md [hunk 2]
```diff
@@ -2,7 +2,7 @@
 code: false
 type: page
 title: Secrets Vault
-description: tbd
+description: Security store your application secrets
```
Comment (scottinet) [inline doc/2/guides/advanced/1-secrets-vault/index.md:5]:
```suggestion
description: Securely store your application secrets
```

File: doc/2/guides/advanced/2-api-keys/index.md [hunk 3]
```diff
@@ -2,9 +2,121 @@
...
 
 # API Keys
 
+Kuzzle allows to create API keys to **authenticate users without using an authentication strategy** and the [auth:login](/core/2/references/controllers/auth/login) action.
+
+An API key is associated with a standard [authentication token](/core/2/guides/main-concepts/5-authentication#authentication-token) that can then be used to [authenticate users](/core/2/api/controllers/auth/login) to the Kuzzle API.
+
+The **authentication tokens** associated to these API keys **may never expire**. It is also possible to **revoke them** at any time by deleting the associated API key.
+
+## Create API Key
```
Comment (scottinet) [inline doc/2/guides/advanced/2-api-keys/index.md:17]:
```suggestion
## Create an API Key
```

File: doc/2/guides/advanced/2-api-keys/index.md [hunk 4]
```diff
@@ -2,9 +2,121 @@
...
+
+</details>
+
+
+Kuzzle returns a response containing the token authentication linked to the API key in the `token` property.
+
+### API Key properties
+
+::: warning
+The authentication token `token` will never be returned by Kuzzle again. If you lose it, you'll have to delete the API key and recreate a new one.
```
Comment (scottinet) [inline doc/2/guides/advanced/2-api-keys/index.md:61]:
```suggestion
The authentication token provided in the `token` property will never be returned by Kuzzle again. If you lose it, you'll have to delete the API key and recreate a new one.
```

File: doc/2/guides/advanced/2-api-keys/index.md [hunk 5]
```diff
@@ -2,9 +2,121 @@
...
+| Property      | Description                                                                       |
+|---------------|-----------------------------------------------------------------------------------|
+| `description` | Description                                                                       |
+| `expiresAt`   | expiration date in UNIX micro-timestamp format (`-1` if the token never expires)  |
+| `fingerprint` | SHA256 hash of the authentication token                                           |
+| `token`       | Authentication token associated with this API key                                 |
+| `ttl`         | Original TTL                                                                      |
+| `userId`      | User [kuid](/core/2/guides/main-concepts/5-authentication#kuzzle-user-identifier) |
+
+## Search for API Key
```
Comment (scottinet) [inline doc/2/guides/advanced/2-api-keys/index.md:73]:
```suggestion
## Search for API Keys
```

File: doc/2/guides/advanced/2-api-keys/index.md [hunk 6]
```diff
@@ -2,9 +2,121 @@
...
+
+It is possible to search in its own API keys ([auth:searchApiKeys](/core/2/references/controllers/auth/search-api-keys)) or in those of all users ([securitys:searchApiKeys](/core/2/references/controllers/securitys/search-api-keys)).
+
+
+In order to know to which API key an authentication token corresponds, it is possible to use the `fingerprint` property which corresponds to the SHA256 hash of the token.
```
Comment (scottinet) [inline doc/2/guides/advanced/2-api-keys/index.md:78]:
```suggestion
In order to know to which API key an authentication token corresponds, it is possible to use the `fingerprint` property which is a SHA256 hash of the token.
```

File: doc/2/guides/advanced/2-api-keys/index.md [hunk 7]
```diff
@@ -2,9 +2,121 @@
...
+
+</details>
+
+::: info
+The associated authentication token is not returned by Kuzzle.
+:::
+
+## Delete API Key
+
+It is possible to use [auth:deleteApiKey](/core/2/references/controllers/auth/delete-api-key) and [security:deleteApiKey](/core/2/references/controllers/security/delete-api-key) methods to delete API keys.
```
Comment (scottinet) [inline doc/2/guides/advanced/2-api-keys/index.md:116]:
```suggestion
It is possible to use the [auth:deleteApiKey](/core/2/references/controllers/auth/delete-api-key) and the [security:deleteApiKey](/core/2/references/controllers/security/delete-api-key) methods to delete API keys.
```

**Review by scottinet — CHANGES_REQUESTED** [review]

**Review by jenow — APPROVED** [review]

File: doc/2/guides/advanced/2-api-keys/index.md [hunk 8]
```diff
@@ -2,9 +2,121 @@
...
+
+An API key is associated with a standard [authentication token](/core/2/guides/main-concepts/5-authentication#authentication-token) that can then be used to [authenticate users](/core/2/api/controllers/auth/login) to the Kuzzle API.
+
+The **authentication tokens** associated to these API keys **may never expire**. It is also possible to **revoke them** at any time by deleting the associated API key.
+
+## Create an API Key
+
+Users can create API keys for their personal use with the [auth:createApiKey](/core/2/references/controllers/auth/create-api-key) action.
+
+Administrators can create API keys for users using the [security:createApiKey](/core/2/references/controllers/security/create-api-key) action.
```
Comment (xbill82) [inline doc/2/guides/advanced/2-api-keys/index.md:21]:
```suggestion
Administrators can create API keys for other users using the [security:createApiKey](/core/2/references/controllers/security/create-api-key) action.
```

**Review by xbill82 — APPROVED** [review]

**Review by scottinet — APPROVED** [review]

**Pull request merged**